### ISSUE TITLE
How to change class balance parameters according to input size

### ISSUE THREAD

**Issue by qimw** [layumi/Seg-Uncertainty#2]
Hi, thank you for your great work on domain adaptation!
The origin input image is 1024*512, but now I want to train on higher resolution, like 1920*1080.
How should I change the class balance parameters like max_value and 64*64*n.
In my opinion, it will be better to use ratio threshold instead of a hard assigned number like 64 like this:
            for i in range(self.num_classes):
                count[i] = torch.sum(labels==i)
                if count[i]/(n*h*w) < 1/128.0: #small classes, original train size is 1024*512
                    weight[i] = self.max_value
and 64*64/(512*1024) = 1/128
But I get worse results than default settings trained with input size (1024*512).
Is there anything that I missed?
Any help will be appreciated.

**Comment by layumi** [comment]
Hi @qimw 
It looks good. You may try to tune the max_value first. 
Different datasets may need different hyper-parameters.

**Comment by qimw** [comment]
Hi @layumi , thank you for your reply. I have a more detailed question.
If I get this kind of pseudo, should I use a larger max_value or a smaller one?
It seems that small objects is good but large object get worse
![GOPR0364_frame_000240_rgb_anon_color](https://user-images.githubusercontent.com/18625505/83316112-b7a2ac00-a256-11ea-9990-36b528427aa9.png)

**Comment by layumi** [comment]
@qimw 
It is hard to say. You may try more regularisation methods, such as auto-augmentation (I wrote in my code but not used in the final model. It sometimes helps. ). 

Besides, you could use the average score of the multi-scale inputs when inference.

**Comment by qimw** [comment]
Thank you for your reply~

**Comment by qimw** [comment]
Hi, I have another problem. Why did you crop images to 512x256 when finetune on pseudo. Is it better than use 1280x640?

**Comment by layumi** [comment]
Due to the GPU memory limitation, this choice is to enable the large batchsize, which could help updating the weights of BatchNorm Layer.

**Comment by qimw** [comment]
Thanks a lot~
